### PR TITLE
feat: add fade transition primitive

### PR DESCRIPTION
**Pull request by martapa**
## Purpose
- Add FadeTransition for showing email submission response
## Checklist

- [ ] Title follows the [naming convention](../README.md#pull-requests).
- [ ] Add a description.
- [ ] Add screenshots (optional).
- [ ] Add testing instructions.
- [ ] Add at least one reviewer.
- [ ] Apply related labels (optional).
- [ ] Leave comments on changes you would like to discuss.
- [ ] Remove changes that are not related to this pull request.
- [ ] Remove debugging code.
- [ ] Test your changes locally and on staging (if applicable).
- [ ] Ensure test coverage value in config matches actual coverage (if applicable).
- [ ] Ensure the build is passing.

### CODE REVIEW

File: src/blocks/FadeTransition.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    },
+  },
+  props: {
+    time: 'string',
+  },
+  state: {},
+  type: 'FadeTransitions',
+}
+
+const FadeTransitions: React.FC<ComposableProps> = ({
```
Comment (mattxyzeth) [inline src/blocks/FadeTransition.tsx]: Name this `FadeTransition` instead. Not plural.

File: src/blocks/FadeTransition.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      args: [['args', 'any'] as ArgTuple],
+      returns: 'void',
+    },
+    hide: {
+      args: [['args', 'any'] as ArgTuple],
+      returns: 'void',
+    },
+  },
+  props: {
+    time: 'string',
```
Comment (mattxyzeth) [inline src/blocks/FadeTransition.tsx:19]: Not sure if this needs to be a number or string for a CSS var. I think a number, but I could be wrong.

Comment (martapa) [reply src/blocks/FadeTransition.tsx:19]: I left it as `string` it didn't work when I tried to concatenate it in the css

File: src/blocks/FadeTransition.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    },
+    hide(args: { [key: string]: any }) {
+      setIsVisible(false)
+    },
+    /* eslint-enable */
+  }
+
+  React.useEffect(() => {
+    if (time) {
+      document.documentElement.style.setProperty('--fade-transition-time', time)
```
Comment (mattxyzeth) [inline src/blocks/FadeTransition.tsx]: let's prefix this with `--bui` so that it's namespaced.

File: src/stories/primitives/FadeTransition.css [hunk 4]
```diff
@@ -0,0 +1,67 @@
+:root {
+  --fade-transition-time: 5s;
+  --fade-animation: fadeIn var(--fade-transition-time)
```
Comment (mattxyzeth) [inline src/stories/primitives/FadeTransition.css]: Should this be a var? Is it configurable by a prop?

File: src/blocks/FadeTransition.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  const actions = {}
+  const [isVisible, setIsVisible] = React.useState(false)
+  const { time } = useProps(context, props)
+  const hooks = {
+    /* eslint-disable */
+    show(args: { [key: string]: any }) {
+      setIsVisible(true)
+    },
+    hide(args: { [key: string]: any }) {
+      setIsVisible(false)
```
Comment (mattxyzeth) [inline src/blocks/FadeTransition.tsx:42]: This doesn't fade the children out, it only removes them. Can you also make it fade out? My suggestion is you have two state variables. One to set visibility and one to add the className. There could be two animation classNames, one for fade-in and one for fade-out.

File: src/stories/primitives/FadeTransition.css (hunk 4 above)
Comment (mattxyzeth) [reply src/stories/primitives/FadeTransition.css]: It's also missing a semi-colon.

File: src/blocks/FadeTransition.tsx (hunk 5 above)
Comment (mattxyzeth) [reply src/blocks/FadeTransition.tsx:42]: fade-out could just reverse the fade-in.

**Review by mattxyzeth — CHANGES_REQUESTED** [review]
couple comments

**Review by mattxyzeth — APPROVED** [review]

**Pull request merged**